store/sql_session_store.go

```diff
@@ -165,6 +165,28 @@ func (me SqlSessionStore) GetSessions(userId string) StoreChannel {
 	return storeChannel
 }
 
+func (me SqlSessionStore) GetSessionsWithActiveDeviceIds(userId string) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+
+	go func() {
+
+		result := StoreResult{}
+		var sessions []*model.Session
+
+		if _, err := me.GetReplica().Select(&sessions, "SELECT * FROM Sessions WHERE UserId = :UserId AND ExpiresAt != 0 AND :ExpiresAt <= ExpiresAt AND LENGTH(DeviceId) > 0", map[string]interface{}{"UserId": userId, "ExpiresAt": model.GetMillis()}); err != nil {
```

Comment:
Is checking `LENGTH(DeviceId)` any different from `DeviceId != ''`? If it's computing the length each time it might be slower and this will get run a lot so might be better to check for empty. Not sure if that's any faster in MySQL and Postgres